src/metabase/driver/oracle.clj

```diff
@@ -0,0 +1,229 @@
+    :minute-of-hour  (hsql/call :extract :minute (hx/->timestamp v))
+    :hour            (trunc :hh v)
+    :hour-of-day     (hsql/call :extract :hour (hx/->timestamp v))
+    :day             (trunc :dd v)
+    ;; subtract number of days between today and first day of week, then add one since first day of week = 1
+    :day-of-week     (hx/inc (hx/- (date :day v)
+                                   (date :week v)))
+    :day-of-month    (hsql/call :extract :day v)
+    :day-of-year     (hx/inc (hx/- (date :day v) (trunc :year v)))
+    ;; [SIC] The format template for truncating to start of week is 'day' in Oracle #WTF
+    :week            (trunc :day v)
+    :week-of-year    (hx/inc (hx// (hx/- (trunc :iw v) ; iw = same day of the week as first day of the ISO year
+                                         (trunc :iy v)) ; iy = ISO year
+                                   7))
+    :month           (trunc :month v)
+    :month-of-year   (hsql/call :extract :month v)
+    :quarter         (trunc :q v)
+    :quarter-of-year (hx// (hx/+ (date :month-of-year (date :quarter v))
+                                 2)
+                           3)
+    :year            (hsql/call :extract :year v)))
+
+(def ^:private ^:const now             (hsql/raw "SYSDATE"))
+(def ^:private ^:const date-1970-01-01 (hsql/call :to_timestamp (hx/literal :1970-01-01) (hx/literal :YYYY-MM-DD)))
+
+(defn- num-to-ds-interval [unit v] (hsql/call :numtodsinterval v (hx/literal unit)))
+(defn- num-to-ym-interval [unit v] (hsql/call :numtoyminterval v (hx/literal unit)))
+
+(defn- date-interval
+  "e.g. (SYSDATE + NUMTODSINTERVAL(?, 'second'))"
+  [unit amount]
+  (hx/+ now (case unit
+              :second  (num-to-ds-interval :second amount)
+              :minute  (num-to-ds-interval :minute amount)
+              :hour    (num-to-ds-interval :hour   amount)
+              :day     (num-to-ds-interval :day    amount)
+              :week    (num-to-ds-interval :day    (hx/* amount (hsql/raw 7)))
+              :month   (num-to-ym-interval :month  amount)
+              :quarter (num-to-ym-interval :month  (hx/* amount (hsql/raw 3)))
+              :year    (num-to-ym-interval :year   amount))))
+
+
+
+(defn- unix-timestamp->timestamp [field-or-value seconds-or-milliseconds]
+  (hx/+ date-1970-01-01 (num-to-ds-interval :second (case seconds-or-milliseconds
+                                                      :seconds      field-or-value
+                                                      :milliseconds (hx// field-or-value (hsql/raw 1000))))))
+
+(defn- apply-offset-and-limit
+  "Append SQL like `OFFSET 20 FETCH FIRST 10 ROWS ONLY` to the query."
+  [honeysql-query offset limit]
+  (assoc honeysql-query
+    :offset (hsql/raw (format "%d ROWS FETCH NEXT %d ROWS ONLY" offset limit) )))
+
+(defn- apply-limit [honeysql-query {value :limit}]
+  ;; Shameless hack! FETCH FIRST ... ROWS ONLY needs to go on the end of the query. Korma doesn't have any built-in
```

Comment:
Actually this is still an issue with HoneySQL but apparently my workaround is the "official" way to handle this situtation:

https://github.com/jkk/honeysql/issues/58#issuecomment-220450400

I'll update this comment since this hack is no longer something I consider "shameless"